src/encoded/audit/genetic_modification.py

```diff
@@ -73,3 +73,28 @@ def audit_genetic_modification_reagents(value, system):
                 method,
             )
             yield AuditFailure('missing genetic modification reagents', detail, level='WARNING')
+
+
+@audit_checker('GeneticModification', frame='object')
+def audit_genetic_modification_reagent_source(value, system):
+    reagents = value.get('reagents', [])
+    for reagent in reagents:
+        source = reagent['source']
+        identifier = reagent['identifier']
+        source_name = source.split('/')[2]
```

Comment:
@caseylitton source.split('/')[2] relies on the identifier structure of source objects, and the embedding of the source object in genetic_modification. I assume @p-sud can add a check that will prevent the exception in case of identifier structure change or embedding change in the future.

Reply:
Yeah I'll change this to get the source object from the using its @id in the reagent, and then use the source's 'name' instead

Reply:
I'm no longer using source_name = source.split('/')[2], I've changed the audit to request.embed the source object then access source.name. source_name_from_identifier is fine as is because it is guaranteed by the schema to include a colon in the middle, and disallow other colons.